Using hasXThrough in conjunction with soft delete

zimudec opened the pull request:
Currently, hasManyThrough allows access to a record from a source table to a target table: table1 -> hasmany -> table_middle -> hasmany -> table2.

It now correctly detects when there are items deleted by soft delete in table_middle.

Due to the wintercms query cache, I had to use "Db::flushDuplicateCache()" during testing to retrieve table1 with the results from table2. I don't know how to avoid this problem.

- Using hasManyThrough in conjunction with soft delete on the intermediate table now respects trashed items
- Testing for hasManyThrough in conjunction with soft delete on the intermediate table

zimudec committed:
- Using hasManyThrough in conjunction with soft delete on the intermediate table now respects trashed items
- Testing for hasManyThrough in conjunction with soft delete on the intermediate table

LukeTowers commented:
@zimudec does this change also need to be applied to hasOneThrough?

bennothommo commented:
@LukeTowers `HasOneThrough` extends `HasManyThrough` :)

LukeTowers commented:
@bennothommo but the original method still exists in HasOneThrough: https://github.com/wintercms/storm/blob/develop/src/Database/Relations/HasOneThrough.php#L35-L41. Should it be removed as a part of this PR?

bennothommo commented:
@LukeTowers ah yeah you're right, must've been looking at Laravel's base relation without realising.

zimudec committed:
- Using hasOneThrough in conjunction with soft delete on the intermediate table now respects trashed items
- Testing for hasOneThrough in conjunction with soft delete on the intermediate table

zimudec commented:
@LukeTowers  I added a change and tests for HasOneThrough as well.

@bennothommo HasOneThrough extends HasManyThrough, but from Eloquent, not from Wintercms. Therefore, I applied the same change to HasOneThrough instead of deleting it.
